layerstack/layer.py

```diff
@@ -416,17 +417,25 @@ class Layer(object):
         Directory from which to load the layer
     """
 
-    def __init__(self, layer_dir):
+    def __init__(self, layer_dir, model=None):
         self.layer_dir = layer_dir
+        self._model = model
         # load the layer.py module and find the LayerBase class
         # self._layer = the LayerBase class we found
         logger.debug(f"Loading layer from {layer_dir}")
         self._layer = self.load_layer(layer_dir)
         self._checksum = checksum(self.layer_filename(layer_dir))
         self._name = self._layer.name
-        self._args = self._layer.args()
+
+        model_layer_base_obj = isinstance(self._layer, ModelLayerBase)
+        if model_layer_base_obj:
+            self._args = self._layer.args(self._model)
+            self._kwargs = self._layer.kwargs(self.model)
+        else:
+            self._args = self._layer.args()
+            self._kwargs = self._layer.kwargs()
```

Comment:
There's no need for the model_layer_base_obj variable here.

```
if isinstance(self._layer, ModelLayerBase):
   self._args = ...
   ...
else:
   ...
```
is just as, if not more, clear.